tests/directory.rs

```diff
@@ -0,0 +1,149 @@
+    let dir = Path::new("~/starship/engine");
+
+    let expected = Segment::new("dir")
+        .set_value("~/starship/engine")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn truncated_directory_in_home() -> io::Result<()> {
+    let dir = Path::new("~/starship/engine/schematics");
+
+    let expected = Segment::new("dir")
+        .set_value("starship/engine/schematics")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn root_directory() -> io::Result<()> {
+    let dir = Path::new("/");
+
+    let expected = Segment::new("dir")
+        .set_value("/")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn directory_in_root() -> io::Result<()> {
+    let dir = Path::new("/private");
+
+    let expected = Segment::new("dir")
+        .set_value("/private")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn truncated_directory_in_root() -> io::Result<()> {
+    let dir = Path::new("/private/var/folders/3s");
+
+    let expected = Segment::new("dir")
+        .set_value("var/folders/3s")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn git_repo_root() -> io::Result<()> {
+    let temp_dir = TempDir::new("starship")?;
+    let repo_dir = temp_dir.path().join("rocket-controls");
+    fs::create_dir(&repo_dir)?;
+
+    Repository::init(&repo_dir).unwrap();
+
+    let expected = Segment::new("dir")
+        .set_value("rocket-controls")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &repo_dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn directory_in_git_repo() -> io::Result<()> {
+    let temp_dir = TempDir::new("starship")?;
+    let repo_dir = temp_dir.path().join("rocket-controls");
+    let dir = repo_dir.join("src");
+    fs::create_dir_all(&dir)?;
+
+    Repository::init(&repo_dir).unwrap();
+
+    let expected = Segment::new("dir")
+        .set_value("rocket-controls/src")
+        .set_style(Color::Cyan.bold())
+        .output();
+    let actual = common::render_segment("dir", &dir);
+    assert_eq!(expected, actual);
+
+    Ok(())
+}
+
+#[test]
+fn truncated_directory_in_git_repo() -> io::Result<()> {
+    let temp_dir = TempDir::new("starship")?;
+    let repo_dir = temp_dir.path().join("rocket-controls");
+    let dir = repo_dir.join("src/meters/fuel-gauge");
+    fs::create_dir_all(&dir)?;
```

Comment:
This is super odd syntax! What does it do? `?;`

Reply:
`let temp_dir = TempDir::new("starship")?;` will compile into:
```rust
match TempDir::new("starship") {
    Ok(dir)  => dir,
    Err(err) => return Err(err),
}
```

It's a quick way to return an `Error` if the function around it is a `Result` (which will allow returning `Error`s).